### ISSUE TITLE
dshackle_request_jsonrpc_err_total doesn't increment on failed requests

### ISSUE THREAD

**Issue by SpontaneousOverthrow** [emeraldpay/dshackle#150]
Dshackle version: from master

Metric `dshackle_request_jsonrpc_err_total` doesn't increment on failed requests(such as method doesn't exist or method is not whitelisted).

**Comment by SpontaneousOverthrow** [comment]
@splix is that correct that `_err` metric should be incremented here?
https://github.com/emeraldpay/dshackle/blob/2b8603df980a7a8707352178c2daf3371a1d3805/src/main/kotlin/io/emeraldpay/dshackle/rpc/NativeCall.kt#L166
https://github.com/emeraldpay/dshackle/blob/2b8603df980a7a8707352178c2daf3371a1d3805/src/main/kotlin/io/emeraldpay/dshackle/rpc/NativeCall.kt#L236
https://github.com/emeraldpay/dshackle/blob/2b8603df980a7a8707352178c2daf3371a1d3805/src/main/kotlin/io/emeraldpay/dshackle/rpc/NativeCall.kt#L243

**Comment by splix** [comment]
Hi @SpontaneousOverthrow, sorry, I've missed the response. Actual metrics depend on the current protocol, so they are handled in BlockchainRpc as `grpc` metrics, and in HttpHandler/WsHandler as `jsonrpc` metrics. So it's expected that there is no error metrics in NativeCall class because it's shared between different protocols.

Anyway, I found out that for HTTP it didn't work for an invalid JSON (as your third example). Fixed it in [eac2bbb]. Also there were issues with Websocket proxy, fixed too, in [10960aa]

Now everything should work as expected, like:
```
# HELP dshackle_request_jsonrpc_err_total Number of requests ended with an error response
# TYPE dshackle_request_jsonrpc_err_total counter
dshackle_request_jsonrpc_err_total{chain="ETH",} 10.0
```

Please let me know if I missed it for other requests

**Comment by SpontaneousOverthrow** [comment]
Everything is fixed now, thanks!
I will close issue a bit later today after some tests.
